web/packages/teleport/src/Main/Main.tsx

```diff
@@ -188,7 +188,7 @@ export function Main(props: MainProps) {
   const banners: BannerType[] = alerts.map(alert => ({
     message: alert.spec.message,
     severity: mapSeverity(alert.spec.severity),
-    link: alert.metadata.labels[LINK_DESTINATION_LABEL],
+    linkDestination: alert.metadata.labels[LINK_DESTINATION_LABEL],
```

Comment:
Could typescript have helped us catch this error?

Reply:
Typescript would have caught this but only if the callback function in the `alerts.map` had a return type defined as suggested in https://github.com/gravitational/teleport/pull/46840#discussion_r1769675377 below.

This is one of the weakness of typescripts' structural type system.

Reply:
Thanks!